lnst/Controller/Controller.py

```diff
@@ -87,7 +96,7 @@ class that implements the
     """
 
     def __init__(self, poolMgr=SlavePoolManager, mapper=MachineMapper,
-                 config=None, pools=[], pool_checks=True, debug=0):
+                 config=None, pools=[], pool_checks=True, debug=0, podman_uri=None, image=None):
```

Comment:
I think the `Controller` shouldn't have the `podman_uri, image` parameters at all, these should go directly to the `PoolManager` class, which also means that the instance needs to be created outside of the Controller class.

So I think we should remove this logic of a *class* as a parameter and instead supply these as instances directly. And if they're None, THEN we create an instance of both a pool manager and a machine mapper, with the default classes.

I think one of the original reasons for this class as a parameter for me was the type hint that it gave, we can now just do that more explicitly via type hints:

```python
def __init__(self, poolMgr: SlavePoolManager = None ...):
    if poolMgr is None:
        poolMgr = SlavePoolManager()
```

Which also probably means that we no longer need the `pool_checks` argument here either and this should be handled manually if needed.

Reply:
It would be quite hard to move initialization of `*PoolManager` outside of `Controller` because `*PoolManager` classes require initialized instances of `MessageDispatcher` and `CtlConfig` whose are initialized in `Controller.__init__()`.  
  
I've replaced `pool_checks`, `podman_uri` and `image` parameters with magic parameter `**poolMgr_kwargs`.